content/articles/automatic-sms-verification-with-sms-retriever-api/index.md

```diff
@@ -0,0 +1,339 @@
+
+Registering and unregistering receivers is usually  done on `onStart()` and `onStop()` methods respectively. The `substringAfterLast()` function is used to extract the code sent through SMS.
+
+> NOTE: Always remember to register and unregister members to avoid memory leaks.
+
+```kotlin
+class MainActivity : AppCompatActivity() {
+    private lateinit var binding: ActivityMainBinding
+    private lateinit var smsClient: SmsRetrieverClient
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivityMainBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        smsClient = SmsRetriever.getClient(this)
+
+        //uncomment this to generate your app hash string. You can view the hash string on your log cat when you run the app
+        /* val appSignatureHelper = SignatureHelper(this)
+        Log.d("SIGNATURE",appSignatureHelper.appSignature.toString())*/
+
+        initSmsListener()
+    }
+    
+     override fun onStart() {
+        super.onStart()
+        EventBus.getDefault().register(this)
+    }
+
+    override fun onStop() {
+        EventBus.getDefault().unregister(this)
+        super.onStop()
+    }
+    
+    private fun initSmsListener() {
+        smsClient.startSmsRetriever()
+            .addOnSuccessListener {
+                Toast.makeText(
+                    this, "Waiting for sms message",
+                    Toast.LENGTH_SHORT
+                ).show()
+            }.addOnFailureListener { failure ->
+                Toast.makeText(
+                    this, failure.localizedMessage,
+                    Toast.LENGTH_SHORT
+                ).show()
+            }
+    }
+
+    @Subscribe
+    fun onReceiveSms(retrievalEvent: RetrievalEvent) {
+        val code: String =
+            StringUtils.substringAfterLast(retrievalEvent.message, "is").replace(":", "")
+                .trim().substring(0, 4)
+
+        runOnUiThread {
+            if (!retrievalEvent.timedOut) {
+                binding.editText.setText(code)
+            } else {
+                Toast.makeText(this, "Failed", Toast.LENGTH_SHORT).show()
+            }
+        }
+        initSmsListener()
+    }
+}
+```
+
+### Computing your app's hash string
+To generate the hash string, you can use the following methods:
+- Use [Play App Signing](https://support.google.com/googleplay/android-developer/answer/9842756?visit_id=637672247631770776-2285078183&rd=1)
+- Use `SignatureHelper class`. This class will help to generate our app's hash string. After using this class to obtain the hash string, always remove it.
+
+```kotlin
+class SignatureHelper(context: Context?) :
+    ContextWrapper(context) {
+    val appSignature: ArrayList<String>
+        get() {
+            val appCodes = ArrayList<String>()
```

Comment:
Hey @robert-muriithi , provide brief comments on what's going on in this class. I think this will shed more light to the reader.